### PR TITLE
Fix PTY scrollback restore with absolute /bin/cat path

### PR DESCRIPTION
**Pull request by daveowenatl**
## Summary

Fixes scrollback restore from PR #161 which was silently failing. The root cause was discovered through telemetry instrumentation: during the first precmd/PROMPT_COMMAND of a freshly-launched amux shell, `\$PATH` doesn't yet include `/bin` or `/usr/bin`. Our amux bin dir is the only entry, so unqualified `cat` and `rm` fail with exit 127.

One-line fix: use absolute paths `/bin/cat` and `/bin/rm` in both zsh and bash integration scripts. This matches cmux's approach and is standard practice for shell integration scripts that run during early shell startup.

## Investigation journey

The scrollback restore mechanism (PR #161) wrote the saved scrollback to a temp file, passed the path via `AMUX_RESTORE_SCROLLBACK_FILE` env var, and expected the shell integration to `cat` the file on startup. It didn't work. Users saw fresh terminals with no restored content.

Initial hypotheses explored and ruled out:
1. **Reader thread timing** — Maybe the shell cat'd the file before amux's reader thread was ready. Added telemetry: reader thread starts before cat runs. Not it.
2. **TTY not attached during .zshenv** — `tty` command reported 'none' when called from .zshenv. Moved integration load from `.zshenv` to later via precmd hook. Still broken.
3. **stdout redirection in precmd** — Maybe zsh captures precmd stdout for prompt building. Added probe: direct `print` calls DID render. Not it.
4. **`/dev/tty` write** — Worked. So the TTY was connected.
5. **`\$(cat ...)` into variable** — Still empty.

Then added stat + ls + `command cat` exit code logging. The telemetry revealed:
```
_amux_restore_scrollback_once:32: command not found: ls
_amux_restore_scrollback_once:33: command not found: wc
_amux_restore_scrollback_once:35: command not found: head
cat exit test: cat fail 127
```

Exit code **127** — command not found. Every single utility was missing from \$PATH because the amux-prepended bin dir was the only entry at that moment.

## Why `/bin/cat` is correct, not a hack

- `/bin/cat` and `/bin/rm` are guaranteed by the POSIX base system layout on macOS, Linux, and BSD
- This is the same approach cmux uses in its shell integration
- Ghostty's own shell integration, iTerm2's, and many others use absolute paths for the same reason
- NixOS is the only edge case (\`/bin/\` is nearly empty) — the `|| true` fallback makes this a silent no-op rather than an error there

## Test plan
- [ ] `cargo clippy --workspace -- -D warnings && cargo fmt --check && cargo test --workspace`
- [ ] Open amux, run a command in a tab, quit, reopen — restored content visible above new prompt
- [ ] Multiple save/restore cycles — content restores correctly each time
- [ ] Works on both zsh and bash

## Related

- Replaces #165 (closed — that PR tried to rework save-side prompt filtering but never got the restore mechanism working; this is the minimal fix that actually unblocks restore)
- Builds on #161 (the PTY-based restore PR that introduced the mechanism)
- Separate issue needed: save-side prompt trimming via Row::semantic_prompt() (still accumulates trailing prompts across cycles)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * More reliable scrollback restore during initial shell startup in Bash and Zsh integrations, preventing failures when the environment is not yet fully initialized while preserving existing error-suppression and cleanup behavior.
  * Improved trimming of trailing prompt lines from captured scrollback so restored/read output no longer includes prompt artifacts.

### CODE REVIEW

File: resources/shell-integration/amux-zsh-integration.zsh [hunk 1]
```diff
@@ -16,8 +16,11 @@ _amux_restore_scrollback_once() {
     unset AMUX_RESTORE_SCROLLBACK_FILE
 
     if [[ -r "$path" ]]; then
-        command cat < "$path" 2>/dev/null || true
-        command rm -f "$path" >/dev/null 2>&1 || true
+        # Use absolute paths: during the first precmd of a freshly-launched
+        # amux shell, $PATH may not yet include /bin or /usr/bin, so
+        # unqualified `cat` fails with "command not found" (exit 127).
```
[the remark this answers is not in the document]
Comment (daveowenatl) [reply resources/shell-integration/amux-zsh-integration.zsh]: Fixed. Reworded to 'during early shell startup' to match the actual execution timing.

File: resources/shell-integration/amux-bash-integration.bash [hunk 2]
```diff
@@ -16,8 +16,11 @@ _amux_restore_scrollback_once() {
     unset AMUX_RESTORE_SCROLLBACK_FILE
 
     if [ -r "$path" ]; then
-        command cat < "$path" 2>/dev/null || true
-        command rm -f "$path" >/dev/null 2>&1 || true
+        # Use absolute paths: during the first PROMPT_COMMAND of a freshly-launched
```
[the remark this answers is not in the document]
Comment (daveowenatl) [reply resources/shell-integration/amux-bash-integration.bash]: Fixed. Same wording update in bash integration.
